File: DataStructure/leetcode1703/greedy.go
// func minMoves(nums []int, k int) int {

// 	p := []int{}

// 	for i, v := range nums {

// 		if v != 0 {

// 			p = append(p, i-len(p))

// 		}

// 	}

// 	m := len(p)

// 	s := make([]int, m+1) // p 的前缀和

// 	for i, v := range p {

// 		s[i+1] = s[i] + v

// 	}

// 	ans := math.MaxInt

// 	for i, v := range s[:m-k+1] { // p[i] 到 p[i+k-1] 中所有数到 p[i+k/2] 的距离之和，取最小值

// 		ans = min(ans, v+s[i+k]-s[i+k/2]*2-p[i+k/2]*(k%2))

// 	}

// 	return ans

// }

File: DataStructure/leetcode1703/greedy.cpp
// class Solution 
// {
// public:
//     int minMoves(vector<int>& nums, int k) 
//     {
//         const int n=nums.size();
//         int i;

//         vector<int>indices;

//         for(i=0;i<n;++i)
//             if(nums[i])indices.emplace_back(i-indices.size());
        
//         const int nOnes=indices.size();

//         vector<int>prefix(nOnes+1);
//         for(i=0;i<nOnes;++i)//prefix[i]表示前i个移动到一起需要的代价
//             prefix[i+1]=prefix[i]+indices[i];
        


//         auto dist_sum=[&](int l,int i,int r)->int
//         {   
//             return prefix[l]+prefix[r]-prefix[i]*2-indices[i]*(k&1);
//         };

//         //枚举左右端点，然后中位数贪心
//         int ans=INT_MAX,left;
//         for(left=0,i=k;i<=nOnes;++left,++i)
//             ans=min(ans,dist_sum(left,(left+i)>>1),i));
        

//         return ans;
//     }

// };